src/sv-pipeline/05_annotation/scripts/compute_AFs.py

```diff
@@ -88,6 +190,12 @@ def calc_allele_freq(record, samples, prefix = None):
             if len([allele for allele in GT if allele != 0 and allele != '.' and allele is not None]) == 2:
                 nhomalt += 1
 
+        #Adjust hemizygous allele number and allele count, if optioned
+        if hemi:
+            AN = AN / 2
+            # For hemizygous sites, AC must be the sum of all non-reference *genotypes*, not alleles
+            AC = nhet + nhomalt
```

Comment:
This is a good point. The problem stems from the fact that our genotyper isn't ploidy-aware (i.e., I don't think we would ever see a `1/.` genotype produced), but we don't want hemizygous sites to count twice towards AC when they're genotyped as homozygous (which I recall happening fairly frequently in XY samples on allosomes).

The solution I've implemented in a revised version of the code (commit pending) is to define `n_gts_with_gt_0_alts` as a designation for the count of genotypes with at least one non-reference allele, and then set `AC` equal to this value for hemizygous sites.

Does that sound like it would handle the edge case you brought up?

Reply:
Sure, that sounds like a good solution! Again, I didn't think this was an actual problem that would cause a bug in the pipeline right now; just that the algorithm for counting alts seemed to be a little inconsistent between the two cases -- which is sometimes the sort of thing that causes bugs down the road.